app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -61,14 +61,19 @@ def allowed_customization_templates(options = {})
     end
   end
 
-  def allowed_hosts_obj(options = {})
-    return [] if (src = resources_for_ui).blank?
-    return [] unless src[:vm]
+  def allowed_datacenters(_options = {})
+    super.slice(datacenter_by_vm.id)
+  end
+
+  def datacenter_by_vm
+    @datacenter_by_vm ||= begin
+                            vm = resources_for_ui[:vm]
+                            VmOrTemplate.find(vm.id).parent_datacenter
```

Comment:
@borod108 Two things:

1) `vm` can be nil when no VM is selected.  This can happen when creating a RHEV catalog item which loads the dialog and shows a list of available templates on the "Catalog" tab.  

Therefore this line should be: `VmOrTemplate.find(vm.id).parent_datacenter if vm`

You access this through Services -> Catalogs -> Catalog Items -> Configuration -> Add a New Catalog Item
Select a "Catalog Item Type": RHEV
Select the tab "Request Info"
![image](https://cloud.githubusercontent.com/assets/2591569/16989535/961fb894-4e62-11e6-9537-b8605c87e7a5.png)


2) Since you are caching `@datacenter_by_vm` and this value can change if the VM selection changes (based on the example in item 1) you will want to clear this value when the VM changes.

The can be achieved by overriding the `set_on_vm_id_changed` method and clearing this value, then calling super.

```ruby
def set_on_vm_id_changed
  @datacenter_by_vm = nil
  super
end
```

cc @syncrou